modules: tfm:Improve TF-M alignment assert message (#10764)
nrfconnect/sdk-nrf: doc/nrf/known_issues.rst

.. rst-class:: v2-3-0

> NCSDK-20864: TF-M unaligned partitions when MCUboot padding and debug optimizations are enabled

Reviewer: Looking at the doc build, I became more aware of the placement of the known issue in this file.
Should it be here, or be moved to the TF-M section?

Author: I think that you are right, I will move it there

> NCSDK-20864: TF-M unaligned partitions when MCUBOOT padding and debug optimizations are enabled. 

Reviewer: ```suggestion
NCSDK-20864: TF-M unaligned partitions when MCUboot padding and debug optimizations are enabled 
```

>   When building TF-M using the partition manager with options :kconfig:option:`CONFIG_BOOTLOADER_MCUBOOT` and :kconfig:option:`CONFIG_DEBUG_OPTIMIZATIONS` enabled, the resulting partitions are not aligned with :kconfig:option:`CONFIG_NRF_SPU_FLASH_REGION_SIZE` as according to :ref:`ug_tfm_partition_alignment_requirements`.

Reviewer: I do not like "as required by the requirements"...
Leaving just "according" does not work quite well either...
How about:
...the resulting partitions are not aligned with :kconfig:option:`CONFIG_NRF_SPU_FLASH_REGION_SIZE` and thus do not comply with the :ref:`ug_tfm_partition_alignment_requirements`.

>   When building TF-M using partition manager with :kconfig:option:`CONFIG_BOOTLOADER_MCUBOOT` and  :kconfig:option:`CONFIG_DEBUG_OPTIMIZATIONS` are enabled the resulted partitions will not be aligned with :kconfig:option:`CONFIG_DEBUG_OPTIMIZATIONS`
>   as required by :ref:`ug_tfm_partition_alignment_requirements` which will cause the build to fail.

Reviewer: ```suggestion
  When building TF-M using the partition manager with options :kconfig:option:`CONFIG_BOOTLOADER_MCUBOOT` and :kconfig:option:`CONFIG_DEBUG_OPTIMIZATIONS` enabled, the resulting partitions are not aligned with :kconfig:option:`CONFIG_DEBUG_OPTIMIZATIONS` as according to :ref:`ug_tfm_partition_alignment_requirements`.
  This will cause the build to fail.
```

Reviewer: We are not mentioning :kconfig:option:`CONFIG_DEBUG_OPTIMIZATIONS` in :ref:`ug_tfm_partition_alignment_requirements`... We should.

Author: That' my mistake actually, the second Kconfig should be the SPU flash size as mentioned in the alignment requirements. I will update this.

>   **Workaround:** Disable :kconfig:option:`CONFIG_DEBUG_OPTIMIZATIONS` or manually edit the partition configuration to follow the requirements in :ref:`ug_tfm_partition_alignment_requirements`.

Reviewer: ```suggestion
  **Workaround:** Disable the :kconfig:option:`CONFIG_DEBUG_OPTIMIZATIONS` option or manually edit the partition configuration to comply with :ref:`ug_tfm_partition_alignment_requirements`.
```

>   **Workaround:** Disable the :kconfig:option:`CONFIG_DEBUG_OPTIMIZATIONS` and :kconfig:option:`CONFIG_TFM_CMAKE_BUILD_TYPE_DEBUG` options or subtract 0x200 from the current value of :kconfig:option:`CONFIG_PM_PARTITION_SIZE_TFM` to comply with :ref:`ug_tfm_partition_alignment_requirements`.

Reviewer: ```suggestion
  **Workaround:** Disable the :kconfig:option:`CONFIG_DEBUG_OPTIMIZATIONS` and :kconfig:option:`CONFIG_TFM_CMAKE_BUILD_TYPE_DEBUG` options, or subtract 0x200 from the current value of the :kconfig:option:`CONFIG_PM_PARTITION_SIZE_TFM` option, to comply with the :ref:`ug_tfm_partition_alignment_requirements`.
```
